### PR TITLE
Add support for comparing files across invocations

### PR DESCRIPTION
**Pull request by siggisim**
Depends on:
- https://github.com/buildbuddy-io/buildbuddy/pull/6588 which adds tabs to the comparison view
- https://github.com/buildbuddy-io/buildbuddy/pull/6592 which allows you to diff files in the editor (which clicking on these rows takes you to)

Screenshot:
Screenshot 2024-05-16 at 4 48 32 PM

### CODE REVIEW

**Review by bduffany — APPROVED** [review]
sweet!!!

nit: IMO this would look slightly nicer with the digest changes right-aligned

File: app/compare/compare_execution_log_files.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    const init = {
+      // Set the stored encoding header to prevent the server from double-compressing.
+      headers: { "X-Stored-Encoding-Hint": "zstd" },
+    };
+
+    this.setState({ loading: true });
+    return rpcService
+      .fetchBytestreamFile(logFile.uri, model.getInvocationId(), "arraybuffer", { init })
+      .then(async (body) => {
+        if (body === null) throw new Error("response body is null");
```
[the remark this answers is not in the document]
Comment (siggisim) [reply app/compare/compare_execution_log_files.tsx]: Done

**Comment by siggisim** [comment]
> sweet!!!
> 
> nit: IMO this would look slightly nicer/ feel a little more digestable (heh) with the digest components right-aligned in each row

Done!

File: app/compare/compare_execution_log_files.tsx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        let entries: tools.protos.ExecLogEntry[] = [];
+        let byteArray = new Uint8Array(body);
+        for (var offset = 0; offset < body.byteLength; ) {
+          let length = varint.decode(byteArray, offset);
```
[the remark this answers is not in the document]
Comment (siggisim) [reply app/compare/compare_execution_log_files.tsx:97]: Haha I looked for those first, and was sad when I saw `--no-delimited` here: https://github.com/buildbuddy-io/protoc-gen-protobufjs?tab=readme-ov-file#protobufjs-cli-compatibility

**Pull request merged**